numpy/lib/npyio.py

```diff
@@ -747,9 +747,9 @@ def _floatconv(x):
     (np.integer, lambda x: int(float(x))),
     (np.longdouble, np.longdouble),
     (np.floating, _floatconv),
-    (complex, lambda x: complex(asstr(x).replace('+-', '-'))),
-    (np.bytes_, asbytes),
-    (np.unicode_, asunicode),
+    (complex, lambda x: complex(x.replace('+-', '-'))),
+    (np.bytes_, methodcaller('encode', 'latin-1')),
+    (np.unicode_, str),
```

Comment:
Is it the case that the input to `_CONVERTERS` is always a `str` and never `bytes`? If so, can you add a comment to `_getconv` saying that the converters are for parsing data from strings?

Reply:
Yes (see also #19609).  Added the comment.